api/app/controllers/api/v1/positions_controller.rb

```diff
@@ -28,6 +28,27 @@ def create
       end
     end
 
+    # POST /positions/import
+    def import
+      @position = Position.new(position_params)
+
+      if @position.valid?
+        if @position.save
+          render json: @position, status: :created
+        else
+          render json: @position.errors, status: :unprocessable_entity
+        end
+      else
+        @already_exists = Position.where('course_code = ? AND session_id = ?', params[:course_code], params[:session_id])
```

Comment:
That makes sense to me, so basically I need another validation before calling update in the duplicate course_code and session id case. I wonder for the update call, you also need to check on which condition you are allowed to update, right? Say if you are failing to satisfy some requirements when you make an update call, that should be automatically caught by the rails validation. If I can use the same mechanism to check before it really updates the data, then I don't need to add the validation here.

Reply:
Yes, sounds like you've got it right. There are a few ways to enforce it, it might need to be mechanical if Rails doesn't provide anything convenient for us to use here. 

Interesting point about update failing. I would still prefer a more mechanical approach to error checking so that we can have a bit more flexibility in the errors we return or how we choose to handle specific cases. Let me know if you need any help moving forward with this !